sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -364,7 +370,7 @@ def assign_rows_csr(X,
     """Densify selected rows of a CSR matrix into a preallocated array.
 
     Like out[out_rows] = X[X_rows].toarray() but without copying.
-    Only supported for dtype=np.float64.
+    No-copy supported for both dtype=np.float32 and dtype=np.float64.
```

Comment:
Maybe it's not clear enough.
Original code will copy `np.float32` data into `np.float64`.
Now it won't copy the data no matter it's  `np.float32` or `np.float64`